### PR TITLE
Release notes for Tarantool releases in 2020-04

### PR DESCRIPTION
**Pull request by xuniq**
fixes gh-2039

### CODE REVIEW

File: doc/release/1.10.10.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    stdin is a tty. Now, ``tarantool -e 'print"Hello"'`` doesn’t enter
+    interactive mode as it was before, just prints ‘Hello’ and exits
+    (:tarantool-issue:`5040`).
+
+..  _build-1:
+
+Build
+~~~~~
+
+-   Make recovering with force_recovery option delete newer than snapshot
```
Comment (NickVolynkin) [inline doc/release/1.10.10.rst:92]:
```suggestion
-   Make recovering with ``force_recovery`` option delete newer than snapshot
```

File: doc/release/2.8.1.rst [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+
+Functionality added or changed
+------------------------------
+
+Core
+~~~~
+
+-   Introduce exclude_null option to index part definition. When it is
+    on, index filters and doesn’t store tuples with “null” value of this
+    part (:tarantool-issue:`4480`).
```
Comment (Totktonada) [inline doc/release/2.8.1.rst:45]: `` :tarantool-issue:`xxxx` `` is rendered as `tarantool/tarantool#xxxx` and there are a lot of such links in release notes. Can we shrink it to just `gh-xxxx` in release notes? We can introduce another clause for the short form.

Comment (NickVolynkin) [reply doc/release/2.8.1.rst:45]: Yes, we can. The template is set in the config https://github.com/tarantool/doc/blob/7fd17c61a920f84f06bd76dbb5dc3dc604633819/conf.py#L40. But what about links to issues in other repos?

Comment (Totktonada) [reply doc/release/2.8.1.rst]: That's why I said about introducing another template rather than modifying this one: it is obvious in this context that the gh-xxxx link means an issue in tarantool. In other contexts we can use the full form (usually the link density is not so tight and the full tarantool/tarantool#xxxx form fits perfectly).

Just like autolinks on GitHub: the link to the same repository is renreder to the short form, while links to another ones are rendered with a full repo name:

```
* https://github.com/tarantool/doc/issues/1
* https://github.com/tarantool/tarantool/issues/1
```

->

* https://github.com/tarantool/doc/issues/1
* https://github.com/tarantool/tarantool/issues/1

File: doc/release/2.8.1.rst [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+    throughput (:tarantool-issue:`5645`).
+
+-   Update operations can’t insert with gaps. This patch changes the
+    behavior so that the update operation fills the missing fields with
+    nulls (:tarantool-issue:`3378`).
+
+-   The new ``box.lib`` module allows loading and executing C
+    stored procedures on read-only nodes (:tarantool-issue:`4642`).
+
+-   Configuration options in `box.cfg` can now be defined with environment variables
```
Comment (Totktonada) [inline doc/release/2.8.1.rst:94]: box.cfg in notable changes and in the list below is wrapped into a span tag rather than italic.

File: doc/release/2.8.1.rst [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+
+Functionality added or changed
+------------------------------
+
+Core
+~~~~
+
+-   The ``exclude_null`` option can now be used in the index part definition.
+    With this option, the index filters and doesn't store tuples with "null" value
+    of the corresponding part part (:tarantool-issue:`4480`).
```
Comment (Totktonada) [inline doc/release/2.8.1.rst:47]: Typo: part part.

File: doc/release/2.8.1.rst [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+
+
+Functionality added or changed
+------------------------------
+
+Core
+~~~~
+
+-   The ``exclude_null`` option can now be used in the index part definition.
+    With this option, the index filters and doesn't store tuples with "null" value
```
Comment (Totktonada) [inline doc/release/2.8.1.rst:46]: Is the comma here intentional? Looks unusual, but I don't know for sure, whether it is incorrect.

File: doc/release/2.8.1.rst [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+~~~~~
+
+-   New tooling for collecting crash artefacts and postmortem analysis
+    (:tarantool-issue:`5569`).
+
+
+Build
+~~~~~
+
+-   Tarantool build infrastructure now requires CMake version 3.1 or later.
```
Comment (Totktonada) [inline doc/release/2.8.1.rst:159]: It is 3.2 due to https://github.com/tarantool/tarantool/pull/5988 (it was pushed to the same set of releases). My bad, I didn't updated the changelog entry.

File: doc/release/2.8.1.rst [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
+
+-   Fixed a double ``gc_cdatanum`` decrementing in LuaJIT platform metrics
+    which occurred when a finalizer was set for a ``GCсdata`` object (:tarantool-issue:`5820`).
+
+
+Lua
+~~~
+
+-   Fixed the `-e` option, when ``tarantool`` used to enter the interactive mode when
+    stdin is a TTY. Now, ``tarantool -e 'print"Hello"'`` doesn't enter the
```
Comment (Totktonada) [inline doc/release/2.8.1.rst:304]: I would use the usual syntax with parenthesis (`tarantool -e "print('Hello')"`).

File: doc/release/2.8.1.rst [hunk 8]
```diff
@@ -0,0 +1,347 @@
...
+
+-   Fix building on FreeBSD (incomplete definition of type ``struct
+    sockaddr``)
+    (:tarantool-issue:`5748`).
+
+-   The already downloaded static build dependencies will not be fetched repeatedly
+    (:tarantool-issue:`5761`).
+
+-   Recovering with ``force_recovery`` option now deletes vylog files which are newer than the snapshot.
+    It helps an instance recover after incidents during a checkpoint (:tarantool-issue:`5823`).
```
Comment (Totktonada) [inline doc/release/2.8.1.rst:340]: Does not look relevant to the build section.

File: doc/release/2.7.2.rst [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+    which could lead to the transaction being applied partially, and to
+    recovery errors. It happened in case the transaction worked with
+    non-sync spaces (:tarantool-issue:`5874`).
+
+-   Fixed a bug in synchronous replication when rolled back transactions
+    could reappear after reconnecting a sufficiently old instance
+    (:tarantool-issue:`5445`).
+
+Module swim
+~~~~~~~~~~~
```
Comment (Totktonada) [inline doc/release/2.7.2.rst:168]: It is just 'Swim' in 2.6.3. And it is usually written in the upper case.

**Comment by Totktonada** [comment]
Development review checklist:

- [ ] 2.8.1
  - [ ] Overview (@Totktonada)
  - [ ] Compatibility (@Totktonada)
  - [ ] Functionality added or changed
    - [ ] Core (@alyapunov or @Korablev77)
    - [ ] Replication (@Gerold103 or @cyrillos)
    - [ ] LuaJIT (@igormunkin)
    - [ ] Tools (@Totktonada)
    - [ ] Build (@Totktonada)
    - [ ] Testing (@Totktonada)
  - [ ] Bugs fixed
    - [ ] Core (@alyapunov or @Korablev77)
    - [ ] Replication (@Gerold103 or @cyrillos)
    - [ ] Module swim (@Gerold103)
    - [ ] LuaJIT (@igormunkin)
    - [ ] Lua (@Totktonada)
    - [ ] SQL (@sergos or @tsafin)
    - [ ] Build (@Totktonada)
- [ ] 2.7.2
  - [ ] Overview (@Totktonada)
  - [ ] Compatibility (@Totktonada)
  - [ ] Functionality added or changed
    - [ ] Core (@alyapunov or @Korablev77)
    - [ ] Replication (@Gerold103 or @cyrillos)
    - [ ] Build (@Totktonada)
    - [ ] Testing (@Totktonada)
  - [ ] Bugs fixed
    - [ ] Core (@alyapunov or @Korablev77)
    - [ ] Replication (@Gerold103 or @cyrillos)
    - [ ] Module swim (@Gerold103)
    - [ ] LuaJIT (@igormunkin)
    - [ ] Lua (@Totktonada)
    - [ ] SQL (@sergos or @tsafin)
    - [ ] Build (@Totktonada)
- [ ] 2.6.3
  - [ ] Overview (@Totktonada)
  - [ ] Compatibility (@Totktonada)
  - [ ] Functionality added or changed
    - [ ] Core (@alyapunov or @Korablev77)
    - [ ] Replication (@Gerold103 or @cyrillos)
    - [ ] Build (@Totktonada)
    - [ ] Testing (@Totktonada)
  - [ ] Bugs fixed
    - [ ] Core (@alyapunov or @Korablev77)
    - [ ] Replication (@Gerold103 or @cyrillos)
    - [ ] Swim (@Gerold103)
    - [ ] Lua (@Totktonada)
    - [ ] SQL (@sergos or @tsafin)
    - [ ] LuaJIT (@igormunkin)
    - [ ] Build (@Totktonada)
- [ ] 1.10.10
  - [ ] Overview (@Totktonada)
  - [ ] Compatibility (@Totktonada)
  - [ ] Functionality added or changed
    - [ ] Build (@Totktonada)
    - [ ] Testing (@Totktonada)
  - [ ] Bugs fixed
    - [ ] Core (@alyapunov or @Korablev77)
    - [ ] Lua (@Totktonada)
    - [ ] Build (@Totktonada)

File: doc/release/2.8.1.rst [hunk 10]
```diff
@@ -0,0 +1,347 @@
...
+    Now each trigger starts in a separate fiber.
+    By default, Tarantool waits for the triggers to finish for 3.0 seconds.
+    User has the option to change this value using the new
+    ``box.ctl.set_on_shutdown_timeout`` function.
+
+    When the timeout expires, Tarantool stops immediately
+    without waiting for the other triggers to finish.
+
+-   Tarantool module developers can now register functions to call when Tarantool stops
+    with the new ``on_shutdown`` API, (:tarantool-issue:`5723`).
```
Comment (Totktonada) [inline doc/release/2.8.1.rst:69]: Forgotten comma?

File: doc/release/2.8.1.rst [hunk 11]
```diff
@@ -0,0 +1,347 @@
...
+    when one of the selected values was ``VARBINARY`` and not directly obtained from the space
+    (:tarantool-issue:`5890`).
+
+
+Build
+~~~~~
+
+-   Fix building on FreeBSD (incomplete definition of type ``struct
+    sockaddr``)
+    (:tarantool-issue:`5748`).
```
Comment (Totktonada) [inline doc/release/2.8.1.rst:334]: Fix -> Fixed (for consistency with other entries).

File: doc/release/2.7.2.rst [hunk 12]
```diff
@@ -0,0 +1,236 @@
...
+    but may trigger elections and become a leader when ``box.ctl.promote()`` is called.
+    When ``election_mode ~= 'manual'``,
+    ``box.ctl.promote()`` replaces ``box.ctl.clear_synchro_queue()``,
+    which is now deprecated.
+
+Build
+~~~~~
+
+
+-   Tarantool build infrastructure now requires CMake version 3.1 or later.
```
Comment (Totktonada) [inline doc/release/2.7.2.rst:64]: (see 2.8.1 comment)

File: doc/release/2.8.1.rst (hunk 10 above)
Comment (NickVolynkin) [reply doc/release/2.8.1.rst:69]: yep

File: doc/release/2.7.2.rst [hunk 13]
```diff
@@ -0,0 +1,236 @@
...
+-   Fixed a double ``gc_cdatanum`` decrementing in LuaJIT platform metrics
+    which occurred when a finalizer was set for a ``GCсdata`` object (:tarantool-issue:`5820`).
+
+Lua
+~~~
+
+-   Fixed the `-e` option, when ``tarantool`` used to enter the interactive mode when
+    stdin is a TTY. Now, ``tarantool -e 'print"Hello"'`` doesn't enter the
+    interactive mode, but just prints "Hello" and exits
+    (:tarantool-issue:`5040`).
```
Comment (Totktonada) [inline doc/release/2.7.2.rst:197]: (see 2.8.1 comment)

File: doc/release/2.7.2.rst [hunk 14]
```diff
@@ -0,0 +1,236 @@
...
+    :tarantool-issue:`5704`, :tarantool-issue:`5913`).
+
+-   Fixed an issue with wrong results of ``SELECT`` with ``GROUP BY`` which occurred
+    when one of the selected values was ``VARBINARY`` and not directly obtained from the space
+    (:tarantool-issue:`5890`).
+
+Build
+~~~~~
+
+-   Fix building on FreeBSD (incomplete definition of type ``struct
```
Comment (Totktonada) [inline doc/release/2.7.2.rst:221]: (see 2.8.1 comment)

File: doc/release/2.7.2.rst [hunk 15]
```diff
@@ -0,0 +1,236 @@
...
+    sockaddr``)
+    (:tarantool-issue:`5748`).
+
+-   The already downloaded static build dependencies will not be fetched repeatedly
+    (:tarantool-issue:`5761`).
+
+-   Recovering with ``force_recovery`` option now deletes vylog files which are newer than the snapshot.
+    It helps an instance recover after incidents during a checkpoint (:tarantool-issue:`5823`).
```
Comment (Totktonada) [inline doc/release/2.7.2.rst:229]: (see 2.8.1 comment)

File: doc/release/2.6.3.rst [hunk 16]
```diff
@@ -0,0 +1,238 @@
...
+    but may trigger elections and become a leader when ``box.ctl.promote()`` is called.
+    When ``election_mode ~= 'manual'``,
+    ``box.ctl.promote()`` replaces ``box.ctl.clear_synchro_queue()``,
+    which is now deprecated.
+
+
+Build
+~~~~~
+
+-   Tarantool build infrastructure now requires CMake version 3.1 or later.
```
Comment (Totktonada) [inline doc/release/2.6.3.rst:67]: (see 2.8.1 comment)

File: doc/release/2.6.3.rst [hunk 17]
```diff
@@ -0,0 +1,238 @@
...
+-   Fixed the crash on attempts to pass an object of a wrong type to
+    ``__serialize`` method of a swim member in Lua (:tarantool-issue:`5952`).
+
+Lua
+~~~
+
+-   Fixed the `-e` option, when ``tarantool`` used to enter the interactive mode when
+    stdin is a TTY. Now, ``tarantool -e 'print"Hello"'`` doesn't enter the
+    interactive mode, but just prints "Hello" and exits
+    (:tarantool-issue:`5040`).
```
Comment (Totktonada) [inline doc/release/2.6.3.rst:191]: (see 2.8.1 comments)

File: doc/release/2.6.3.rst [hunk 18]
```diff
@@ -0,0 +1,238 @@
...
+-   Fixed a double ``gc_cdatanum`` decrementing in LuaJIT platform metrics
+    which occurred when a finalizer was set for a ``GCсdata`` object (:tarantool-issue:`5820`).
+
+
+Build
+~~~~~
+
+-   Fix building on FreeBSD (incomplete definition of type ``struct
+    sockaddr``)
+    (:tarantool-issue:`5748`).
```
Comment (Totktonada) [inline doc/release/2.6.3.rst:225]: (see 2.8.1 comment)

File: doc/release/2.6.3.rst [hunk 19]
```diff
@@ -0,0 +1,238 @@
...
+
+-   The already downloaded static build dependencies will not be fetched repeatedly
+    (:tarantool-issue:`5761`).
+
+-   Recovering with ``force_recovery`` option now deletes vylog files which are newer than the snapshot.
+    It helps an instance recover after incidents during a checkpoint (:tarantool-issue:`5823`).
```
Comment (Totktonada) [inline doc/release/2.6.3.rst:231]: (see 2.8.1 comment)

File: doc/release/1.10.10.rst [hunk 20]
```diff
@@ -0,0 +1,92 @@
...
+using the ``box.schema.upgrade()`` procedure to unlock all the new
+features of the 1.10.x series.
+
+Functionality added or changed
+------------------------------
+
+Build
+~~~~~
+
+-   Tarantool build infrastructure now requires CMake version 3.1 or later.
```
Comment (Totktonada) [inline doc/release/1.10.10.rst:41]: (see 2.8.1 comment)

File: doc/release/1.10.10.rst [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+    and for some functions from the modules ``fio``, ``box.tuple``, and ``iconv``
+    (:tarantool-issue:`5632`).
+
+Lua
+~~~
+
+-   Fixed the `-e` option, when ``tarantool`` used to enter the interactive mode when
+    stdin is a TTY. Now, ``tarantool -e 'print"Hello"'`` doesn't enter the
+    interactive mode, but just prints "Hello" and exits
+    (:tarantool-issue:`5040`).
```
Comment (Totktonada) [inline doc/release/1.10.10.rst:85]: (see 2.8.1 comments)

File: doc/release/1.10.10.rst [hunk 22]
```diff
@@ -0,0 +1,92 @@
...
+
+
+Build
+~~~~~
+
+-   Recovering with ``force_recovery`` option now deletes vylog files newer than the snapshot.
+    It helps an instance recover after incidents during a checkpoint (:tarantool-issue:`5823`).
```
Comment (Totktonada) [inline doc/release/1.10.10.rst:92]: (see 2.8.1 comments)

File: doc/release/2.8.1.rst (hunk 7 above)
Comment (NickVolynkin) [reply doc/release/2.8.1.rst:304]: what's the meaning of parentheses?

Comment (Totktonada) [reply doc/release/2.8.1.rst:304]: A function call. I don't know why print works without parentheses. It seems, it is some weird exception from usual syntax rules.

File: doc/release/2.8.1.rst [hunk 23]
```diff
@@ -0,0 +1,347 @@
...
+*   Release :tarantool-release:`2.8.1`.
+*   Tagged ``2.8.1-0-ge2a1ec0``
+
+Overview
+--------
+
+2.8.1 is the :doc:`beta </release/policy>`
+version of the 2.8 release series.
+
+This release introduces 28 new features and resolves 31 bug since version :doc:`2.7.2 </release/2.7.2>`.
```
Comment (modronmarch) [inline doc/release/2.8.1.rst:15]: "since version :doc:`2.7.2" выглядит калькой с русского. я бы сформулировал как "compared with version :doc:`2.7.2"

File: doc/release/2.8.1.rst [hunk 24]
```diff
@@ -0,0 +1,347 @@
...
+There can be bugs in less common areas.
+If you find any, feel free to `report an
```
Comment (modronmarch) [inline doc/release/2.8.1.rst:17]: "feel free" - это "не стесняйтесь". м.б. что-нибудь типа "please help us improve by reporting an issue?"

File: doc/release/2.8.1.rst [hunk 25]
```diff
@@ -0,0 +1,347 @@
...
+issue <https://github.com/tarantool/tarantool/issues>`__ on GitHub.
+
+Notable changes are:
```
Comment (modronmarch) [inline doc/release/2.8.1.rst:20]: я бы привел все четыре пункта к общему знаменателю, иначе мне было немного сложно читать и понимать, что конкретно было сделано

* Added the ability to set multiple iproto threads.
* Added the ability to set ``box.cfg`` options with environment variables.
* Implemented the new ``box.ctl.promote()`` function and introduced the concept of manual elections.
* Made some enhancements to the Lua memory profiler.

File: doc/release/2.8.1.rst [hunk 26]
```diff
@@ -0,0 +1,347 @@
...
+-  Tarantool is now able to set multiple iproto threads.
+-  Setting ``box.cfg`` options with environment variables.
+-  The new ``box.ctl.promote()`` function and the concept of manual elections.
+-  Enhancements in the Lua memory profiler.
+
+Compatibility
+-------------
+
+Tarantool 2.x is backward compatible with Tarantool 1.10.x in the binary
+data layout, client-server protocol, and replication protocol.
```
Comment (modronmarch) [inline doc/release/2.8.1.rst:31]: Tarantool 2.x binary data layout, client-server protocol, and replication protocol are backward compatible with Tarantool 1.10.x ?

File: doc/release/2.8.1.rst [hunk 27]
```diff
@@ -0,0 +1,347 @@
...
+
+Please
```
Comment (modronmarch) [inline doc/release/2.8.1.rst:33]: Широко используемая практика - использовать please только в случаях, где мы доставляем человеку неудобство (please wait, please try again). слово теряет смысл, если его слишком часто использовать.

Comment (modronmarch) [reply doc/release/2.8.1.rst:33]: м.б. начать с call to action? 
To unlock all the new features of the 2.x series, `upgrade <https://www.tarantool.io/en/doc/latest/book/admin/upgrades/>`__
using the ``box.schema.upgrade()`` procedure.

File: doc/release/2.8.1.rst [hunk 28]
```diff
@@ -0,0 +1,347 @@
...
+    *   default values,
+
+
+Replication
+~~~~~~~~~~~
+
+-   Introduced the ``box.ctl.promote()`` function and the concept of manual elections
+    (enabled with ``election_mode='manual'``) (:tarantool-issue:`3055`).
+
+    Once the instance is in the ``manual`` election mode,
```
Comment (modronmarch) [inline doc/release/2.8.1.rst:111]: the instance > an instance? она одна неповторимая или их может быть несколько?

File: doc/release/2.8.1.rst [hunk 29]
```diff
@@ -0,0 +1,347 @@
...
+    it acts like a ``voter`` most of the time,
+    but may trigger elections and become a leader when ``box.ctl.promote()`` is called.
+    When ``election_mode ~= 'manual'``,
```
Comment (modronmarch) [inline doc/release/2.8.1.rst:114]: это предложение пришлось несколько раз прочитать, пока не заметил запятую
м.б. ``box.ctl.promote()`` replaces ``box.ctl.clear_synchro_queue()`` (which is now deprecated) if ``election_mode ~= 'manual'``.

File: doc/release/2.8.1.rst [hunk 30]
```diff
@@ -0,0 +1,347 @@
...
+-   Tarantool build infrastructure now requires CMake version 3.1 or later.
+
+-   Binary packages for Fedora 33 are now available (:tarantool-issue:`5502`) .
+
+-   Binary packages for CentOS 6 and Debian Jessie won't be published since this version.
+
+-   RPM and DEB packages no longer have the ``autotools`` dependency (follows up
+    :tarantool-issue:`4968`).
+
+-   Regular testing on MacOS 10.13 has been disabled, effectively stopping
```
Comment (modronmarch) [inline doc/release/2.8.1.rst:168]: disabled = отключен. неестественно звучит, либо я не в контексте.
м.б. discontinued или We no longer test on MacOS 10.13 regularly ?

File: doc/release/2.8.1.rst [hunk 31]
```diff
@@ -0,0 +1,347 @@
...
+
+-   The ``libcurl`` headers are now shipped to system path ``${PREFIX}/include/tarantool``
+    when ``libcurl`` is included as a bundled library or in a static build
+    (:tarantool-issue:`4559`).
+
+
+Testing
+~~~~~~~
+
+-   Tarantool CI/CD has migrated to GitHub Actions (:tarantool-issue:`5662`).
```
Comment (modronmarch) [inline doc/release/2.8.1.rst:184]: was migrated?

Comment (modronmarch) [reply doc/release/2.8.1.rst:184]: он же не сам это сделал, полагаю )

File: doc/release/2.8.1.rst [hunk 32]
```diff
@@ -0,0 +1,347 @@
...
+
+-   Fixed the dirty read after restart while using MVCC with synchronous
+    replication (:tarantool-issue:`5973`).
+
+
+Replication
+~~~~~~~~~~~
+
+-   Fixed an issue with an applier hanging on a replica after failing to process
+    a ``CONFIRM`` or ``ROLLBACK`` message coming from a master.
```
Comment (modronmarch) [inline doc/release/2.8.1.rst:254]: a master > the master? может быть больше одного мастера?

Comment (Gerold103) [reply doc/release/2.8.1.rst:254]: Во-первых, да. Если сплит бреин был, то какое-то время их может быть два, хотя один не сможет ничего делать. Во-вторых, тут скорее смысл, что это где-то какой-то один мастер обычно, "некий", на неопределенной удаленной машине. Не какой-то конкретный.
